### ISSUE TITLE
transformStyles not working correctly

### ISSUE THREAD

**Issue by pascalknecht** [WordPress/gutenberg#16408]
**Describe the bug**
When loading custom css via add_editor_styles not every style is applied when any element has an svg background encoded as a data:image/svg.

**To reproduce**
1. Enter the following code into the console
`
wp.editor.transformStyles([{css: ".navbar-light .navbar-toggler-icon {↵background-image: url(\"data:image/svg+xml;charset=utf8,%3Csvg viewBox='0 0 30 30' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke='rgba(0, 0, 0, 0.5)' stroke-width='2' stroke-linecap='round' stroke-miterlimit='10' d='M4 7h22M4 15h22M4 23h22'/%3E%3C/svg%3E\")}"}], '.editor-styles-wrapper');
`

**Expected behavior**
The styles should be transformed with `.editor-styles-wrapper` in front.

**Additional context**
- Gutenberg 6.0.0

**Comment by oxyc** [comment]
I just hit this as well when having having swiper included. It breaks at https://github.com/nolimits4web/swiper/blob/1c4dfa3b08727f7febd9238980d8249d89202702/dist/css/swiper.css#L490

**Comment by noisysocks** [comment]
This seems to be working for me. When I paste the snippet into my console I get:

```
[
	".editor-styles-wrapper .navbar-light .navbar-toggler-icon { background-image: url("data:image/svg+xml;charset=utf8,%3Csvg viewBox='0 0 30 30' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke='rgba(0, 0, 0, 0.5)' stroke-width='2' stroke-linecap='round' stroke-miterlimit='10' d='M4 7h22M4 15h22M4 23h22'/%3E%3C/svg%3E"); }"
]
```

**Comment by oxyc** [comment]
@noisysocks on slack you mentioned you had to remove the linebreak to make it work. That's part of the bug. See the upstream bug report I mentioned in https://github.com/WordPress/gutenberg/issues/16408#issuecomment-523862740.

I have a working PR for this https://github.com/WordPress/gutenberg/pull/17146

**Comment by noisysocks** [comment]
Thanks @oxyc—you're totally right and that second example there demonstrates the bug when I run it locally.

**Comment by strarsis** [comment]
Similar issue: https://github.com/WordPress/gutenberg/issues/21569
Relevant PR: https://github.com/WordPress/gutenberg/pull/21936

**Comment by strarsis** [comment]
New editor styles transformation approach using [PostCSS](https://github.com/WordPress/gutenberg/pull/49521) is now merged (many thanks @zaguiini)!

**Comment by skorasaurus** [comment]
This is now fixed thanks to the work of @zaguiini and @strarsis ! 

This will be included in gutenberg 17.0; was specifically https://github.com/WordPress/gutenberg/pull/49521 

before https://github.com/WordPress/gutenberg/pull/49521 

![image](https://github.com/WordPress/gutenberg/assets/955351/d664647e-61ee-44fa-8598-04a655ccf457)

after: 
![image](https://github.com/WordPress/gutenberg/assets/955351/7c667292-205f-444a-a065-daab92249529)

@oxyc I've also confirmed that the merged PR fixes your reduced test cases as well.
